networkmanager/ipamalloc.go

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright (C) 2023 Renesas Electronics Corporation.
+// Copyright (C) 2023 EPAM Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package networkmanager provides set of API to configure network
+
+package networkmanager
+
+import (
+	"net"
+	"sync"
+
+	"github.com/aoscloud/aos_common/aoserrors"
+	"github.com/apparentlymart/go-cidr/cidr"
+	log "github.com/sirupsen/logrus"
+)
+
+/***********************************************************************************************************************
+ * Types
+ **********************************************************************************************************************/
+
+type subnetwork struct {
+	ipNet *net.IPNet
+	ips   []net.IP
+}
+
+type ipSubnet struct {
+	sync.Mutex
+	predefinedPrivateNetworks []*net.IPNet
+	usedIPSubnetNetworks      map[string]subnetwork
+}
+
+/***********************************************************************************************************************
+ * Private
+ **********************************************************************************************************************/
+
+func newIPam() (ipam *ipSubnet, err error) {
+	log.Debug("Create ipam allocator")
+
+	ipam = &ipSubnet{}
+
+	if ipam.predefinedPrivateNetworks, err = makeNetPools(); err != nil {
+		return nil, err
+	}
+
+	ipam.usedIPSubnetNetworks = make(map[string]subnetwork)
+
+	return ipam, nil
+}
+
+func (ipam *ipSubnet) tryToGetExistIPNetFromPool(networkID string) (*net.IPNet, bool) {
```

Comment:
`getIPNetFromPool`, `getExistIPNet`?
Why do need this function at all?

Reply:
removed